src/main/connection.cpp

```diff
@@ -30,7 +30,21 @@ Connection::Connection(DatabaseInstance &database) : context(make_shared<ClientC
 Connection::Connection(DuckDB &database) : Connection(*database.instance) {
 }
 
+Connection::Connection(Connection &&other) noexcept {
+	std::swap(context, other.context);
+	std::swap(warning_cb, other.warning_cb);
```

Comment:
`warning_cb` does not seem to be used anywhere AFAICS (same goes for `SetWarningCallback`, which is unimplemented). Maybe remove that field / method?